src/codegen.cpp

```diff
@@ -2625,12 +2651,13 @@ static Value *emit_f_is(const jl_cgval_t &arg1, const jl_cgval_t &arg2, jl_codec
         Value *varg2 = arg2.constant ? literal_pointer_val(arg2.constant) : arg2.V;
         assert(varg1 && varg2 && (arg1.isboxed || arg1.TIndex) && (arg2.isboxed || arg2.TIndex) &&
                 "Only boxed types are valid for pointer comparison.");
-        return builder.CreateICmpEQ(varg1, varg2);
+        return builder.CreateICmpEQ(decay_derived(varg1),
+                                    decay_derived(varg2));
     }
 
     JL_FEAT_REQUIRE(ctx, runtime);
-    Value *varg1 = boxed(arg1, ctx);
-    Value *varg2 = boxed(arg2, ctx, false); // potentially unrooted!
+    Value *varg1 = mark_callee_rooted(boxed(arg1, ctx));
```

Comment:
That's fine, the pass will still keep it live at any safepoint that it's life *across*. It just won't put it in a gc root if a safepoint terminates the live interval.